src/hooks/useIsOnScreen.ts

```diff
@@ -0,0 +1,41 @@
+import { RefObject, useEffect, useRef, useState } from 'react';
+
+export const useIsOnScreen = (ref: RefObject<HTMLElement>) => {
+  const observerRef = useRef<IntersectionObserver | null>(null);
+  const [isOnScreen, setIsOnScreen] = useState(true);
+
+  useEffect(() => {
+    if (ref.current) {
+      const scrollParent = getScrollParent(ref.current);
+      observerRef.current = new IntersectionObserver(
```

Comment:
```suggestion
      const observer = new IntersectionObserver(
```
we don't really need that `observerRef` do we? The intersection observer doesn't leak outside of the effect hook.

Reply:
good call, done